Trace Analytics support for custom sources

ps48 opened the pull request:
### Description
Experimental feature for Trace Analytics support of custom sources

### Issues Resolved
#2052 

* Update mode picker with custom trace source 
<img width="1792" alt="Screenshot 2024-09-03 at 5 33 59 PM" src="https://github.com/user-attachments/assets/4a2be5a7-5ab1-4f43-b790-48f2f0c13be2">


* Flyout to manage/update custom trace source
<img width="1792" alt="Screenshot 2024-09-03 at 5 34 07 PM" src="https://github.com/user-attachments/assets/00c4bb87-9fcf-433e-bb96-954d7e1f5f97">

* Add service filtering widgets to traces view
<img width="1792" alt="Screenshot 2024-09-03 at 5 34 24 PM" src="https://github.com/user-attachments/assets/a7d9032d-501e-415c-bc3c-b825792cb1c7">

* Lazy load red metrics 
<img width="1792" alt="Screenshot 2024-09-03 at 5 34 52 PM" src="https://github.com/user-attachments/assets/3c4f5e12-915b-4a36-92f1-272318c6996f">


* Hide/Show attribute columns
<img width="1792" alt="Screenshot 2024-09-03 at 5 36 01 PM" src="https://github.com/user-attachments/assets/7d27fff7-b1fa-431c-82b8-8b3cbdaa3712">

* Added custom trace indices in user settings 
<img width="1792" alt="Screenshot 2024-09-03 at 5 37 43 PM" src="https://github.com/user-attachments/assets/16ff73d0-8507-48cb-a97d-ee831f667692">


### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

TackAdam commented on public/components/trace_analytics/components/common/plots/service_map.tsx:
```diff
@@ -93,6 +97,35 @@ export function ServiceMap({
+        includeMetricsCallback();
+      }
+    }
+    // console.log('value[0].id', value);
```
Missed deleting console log comment.

TackAdam commented:
UI looks correct / functional under the new navigation.
<img width="1743" alt="Screenshot 2024-09-03 at 6 41 35 PM" src="https://github.com/user-attachments/assets/be79874a-a41b-4f8b-bc62-9b1bc46928a5">

TackAdam commented:
<img width="546" alt="Screenshot 2024-09-03 at 6 35 58 PM" src="https://github.com/user-attachments/assets/cf5137ba-9397-4e28-9beb-0001a163bbec">

Under the new navigation when services and traces are broken apart, do we need to change this drop down for services?

TackAdam commented:
<img width="1197" alt="Screenshot 2024-09-03 at 6 44 08 PM" src="https://github.com/user-attachments/assets/8754f57b-953e-47e4-b98c-b0efc01752b4">

Spacing seems a little off between the drop downs. Not sure if we have a reference for multiple bars.
Not blocking.

TackAdam reviewed (approved):
Great work!

ps48 commented:
@TackAdam @RyanL1997  Thanks for the review. 
> Spacing seems a little off between the drop downs. Not sure if we have a reference for multiple bars.
* Updated spacing for flex items
<img width="1216" alt="Screenshot 2024-09-03 at 9 42 05 PM" src="https://github.com/user-attachments/assets/b68ed56b-aff3-4336-b3f8-b64ee511eef6">


> Under the new navigation when services and traces are broken apart, do we need to change this drop down for services?
* Changed it to custom source
